tensorflow/lite/micro/memory_planner/greedy_memory_planner.h

```diff
@@ -110,16 +118,23 @@ class GreedyMemoryPlanner : public MemoryPlanner {
   // Records the client-provided information about each buffer.
   struct BufferRequirements {
     int size;
+    int offline_offset;
     int first_time_used;
     int last_time_used;
   };
 
   // Working arrays used during the layout algorithm.
   BufferRequirements* requirements_;
-  int* buffer_sizes_sorted_by_size_;
-  int* buffer_ids_sorted_by_size_;
+  // buffer_sizes_sorted_ and buffer_ids_sorted_ are sorted according to:
+  //   {
+  //     offline planned buffers,
+  //     online planned buffers sorted by size
+  //   }
+  int* buffer_sizes_sorted_;
+  int* buffer_ids_sorted_;
   ListEntry* buffers_sorted_by_offset_;
   int next_free_entry_;
+  int first_entry_index_;
```

Comment:
In previous implementation the smallest buffer offset was always located at first in the list (buffers_sorted_by_offset_[0]). This may not be the case when offline planned tensors are present. Therefore, first_entry_index_, keeps track of what index in the list has the smallest buffer offset.